File: backend/src/main/java/com/kiwi/features/skills/data/SkillDTO.java
package com.kiwi.features.skills.data;

import lombok.*;
import java.time.Instant;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder @EqualsAndHashCode @ToString
public class SkillDTO {

    private Long skillId;
    private String name;
    private String description;
    private String quote;
    private int icon;

    private String cooldownType;
    private Long cooldownGoalId;
    private Integer cooldownTimeMinutes;
    private String cooldownOtherDescription;
    private Long levelupSkillId;

    private boolean isCooldown;
    private Long cooldownUntil;
    private int deckSlot;
}

File: mobile/app/src/main/java/com/bellako/kiwi/features/skills/data/SkillDTO.kt
package com.bellako.kiwi.features.skills.data

data class SkillDTO(
    val skillId: Long,
    val name: String,
    val type: String,
    val description: String,
    val quote: String?,
    val cooldownType: String,
    val cooldownGoalId: Long?,
    val cooldownTimeMinutes: Int?,
    val cooldownOtherDescription: String?,
    val levelupSkillId: Long?,
    val cooldown: Boolean,
    val cooldownUntil: Long?,
    val deckSlot: Int,
)

enum class CooldownType {
    TIME,
    OTHER,
    GOAL,
}

enum class SkillType {
    ADAPTABILITY,
    CONTROL,
    EMPATHY,
    FOCUS,
    MOTIVATION,
    RESILIENCE,
}
